pallets/phala/src/migrations/mod.rs

```diff
@@ -120,3 +140,263 @@ pub mod v6 {
 		Ok(())
 	}
 }
+
+pub mod stakepoolv2_migration {
+	use super::*;
+
+	const STAKING_ID: LockIdentifier = *b"phala/sp";
+	const VESTING_ID: LockIdentifier = *b"vesting ";
+	const DEMOCRAC_ID: LockIdentifier = *b"democrac";
+	const PHRELECT_ID: LockIdentifier = *b"phrelect";
+	fn migrate_stake_pools<T: PhalaPallets>()
+	where
+		T: PhalaPallets + stakepool::pallet::Config + PhalaConfig,
+		BalanceOf<T>: balance_convert::FixedPointConvert + sp_std::fmt::Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32>,
+		T: pallet_assets::Config<Balance = BalanceOf<T>>,
+	{
+		stakepool::pallet::StakePools::<T>::iter().for_each(|(pid, pool_info)| {
+			let collection_id: CollectionId = pallet_rmrk_core::Pallet::<T>::collection_index();
+			let symbol: BoundedVec<u8, <T as pallet_rmrk_core::Config>::CollectionSymbolLimit> =
+				format!("STAKEPOOL-{}", pid)
+					.as_bytes()
+					.to_vec()
+					.try_into()
+					.expect("create a bvec from string should never fail; qed.");
+			pallet_rmrk_core::Pallet::<T>::create_collection(
+				Origin::<T>::Signed(basepool::pallet_id::<T::AccountId>()).into(),
+				Default::default(),
+				None,
+				symbol,
+			)
+			.expect("create collection should not fail; qed.");
+			let account_id = basepool::pallet::create_staker_account::<T::AccountId>(
```

Comment:
Does `create_staker_account` just generate an account? If so, better to rename to `generate_stake_account` instead of "create_", since it doesn't write to any chain state.

Reply:
done